packages/python/accern_data/accern_data.py

```diff
@@ -425,41 +528,59 @@ def _process_date(
             output_path: str,
             output_pattern: Optional[str],
             *,
-            is_first_day: bool,
-            progress_bar: ProgressBar) -> None:
+            is_first_time: bool,
+            mode: Mode,
+            filters: Dict[str, str],
+            progress_bar: ProgressBar) -> bool:
         self._params["date"] = cur_date
         first = True
-        for res in self._scroll("1900-01-01"):
+        for res in self._scroll("1900-01-01", mode, filters):
             is_empty = False
             if first:
-                self.get_mode().init_day(
-                    cur_date, output_path, output_pattern, is_first_day)
+                mode.init_day(
+                    cur_date, output_path, output_pattern, is_first_time)
                 first = False
+
             if isinstance(res, pd.DataFrame) and res.empty:
                 is_empty = True
             elif isinstance(res, list) and len(res) == 0:
                 is_empty = True
+
+            if is_first_time and not is_empty:
+                is_first_time = False
+
             if not is_empty:
-                self.get_mode().add_result(res)
+                mode.add_result(res)
                 progress_bar.update(len(res))
 
         if not first:
-            self.get_mode().finish_day()
+            mode.finish_day()
+        return is_first_time
 
     def download_range(
             self,
             start_date: str,
             output_path: Optional[str] = None,
             output_pattern: Optional[str] = None,
             end_date: Optional[str] = None,
+            mode: Optional[ModeType] = None,
+            split_dates: bool = True,
+            filters: Optional[FiltersType] = None,
             verbose: bool = False) -> None:
         global VERBOSE
         VERBOSE = verbose
         if output_path is None:
             output_path = "./"
         os.makedirs(output_path, exist_ok=True)
+        valid_mode = self.get_mode() \
+            if mode is None else self.parse_mode(mode, split_dates)
+        if filters is None:
+            valid_filters = self.get_filters()
+        else:
+            valid_filters = self.parse_filters(self.validate_filters(filters))
```

Comment:
Filters updates the already existing filter. Be aware of none which is supposed to remove a filter